File: 217.contains-duplicate.py
#
# @lc app=leetcode id=217 lang=python
#
# [217] Contains Duplicate
#

# @lc code=start
class Solution(object):
    def containsDuplicate(self, nums):
        num_set = set()
        for num in nums:
            if num in num_set:
                return True
            num_set.add(num)
        
        return False
        
# @lc code=end

File: 217.contains-duplicate.rs
impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut num_set: HashSet<i32> = HashSet::new();

        for num in nums{
            if num_set.contains(&num){
                return true
            }
            num_set.insert(num);
        }
        false
    }
}